### ISSUE TITLE
add support for CAA DNS records to aws_route53_record resource

### ISSUE THREAD

**Issue by hashibot** [terraform-providers/terraform-provider-aws#1632]
_This issue was originally opened by @danielnbarbosa as hashicorp/terraform#16054. It was migrated here as a result of the [provider split](https://www.hashicorp.com/blog/upcoming-provider-changes-in-terraform-0-10/). The original body of the issue is below._

<hr>

The new CAA record is supported by AWS Route53 but not considered a valid type by terraform.  CAA is used to whitelist which certificate authorities are authorized to issue SSL certificates for your domain.  It is quite new and was only added to AWS on August 21, 2017.  CAs are expected to start enforcing it in September 2017.

### Terraform Version
Terraform v0.10.4

### Terraform Configuration Files
```hcl
resource "aws_route53_record" "pear_caa" {
  zone_id = "${aws_route53_zone.pear.zone_id}"
  name    = "pear.md"
  type    = "CAA"
  ttl     = "300"
  records = ["0 issue \"godaddy.com\""]
}
```

### Expected Behavior
terraform plan should work

### Actual Behavior
terraform plan returns error on resource type:
```
1 error(s) occurred:

* module.global_prod.aws_route53_record.pear_caa: "type" must be one of [SOA, A, TXT, NS, CNAME, MX, NAPTR, PTR, SRV, SPF, AAAA]
```

### Steps to Reproduce
Please list the full steps required to reproduce the issue, for example:
1. `terraform plan`

- hashicorp/terraform#6017

**Comment by bflad** [comment]
@danielnbarbosa this support was merged into master a few weeks ago via #1465. However, it has not been "released" yet via version 0.1.5, 0.2.0, or 1.0.0 given the current [CHANGELOG](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md). You may wish to participate in #1592 asking for releases of this provider more often if its something you're wanting something sooner rather than later.

**Comment by stack72** [comment]
@Ninir I believe this can be closed via #1465

**Comment by Ninir** [comment]
Hi folks,

This was merged in https://github.com/terraform-providers/terraform-provider-aws/pull/1467.
Closing!

(Thanks for the ping @stack72 ! 👍 )
